src/plugins/dashboard/public/services/dashboard_content_management/lib/find_dashboards.ts

```diff
@@ -83,19 +83,34 @@ export async function findDashboardById(
       references: cachedDashboard.item.references,
     };
   }
+
   /** Otherwise, fetch the dashboard from the content management client, add it to the cache, and return the result */
-  const response = await contentManagement.client
-    .get<DashboardCrudTypes['GetIn'], DashboardCrudTypes['GetOut']>({
+  try {
+    const response = await contentManagement.client.get<
+      DashboardCrudTypes['GetIn'],
+      DashboardCrudTypes['GetOut']
+    >({
       contentTypeId: DASHBOARD_CONTENT_ID,
       id,
-    })
-    .then((result) => {
-      dashboardContentManagementCache.addDashboard(result);
-      return { id, status: 'success', attributes: result.item.attributes };
-    })
-    .catch((e) => ({ status: 'error', error: e.body, id }));
+    });
+    if (response.item.error) {
+      throw response.item.error;
+    }
 
-  return response as FindDashboardsByIdResponse;
+    dashboardContentManagementCache.addDashboard(response);
+    return {
+      id,
+      status: 'success',
+      attributes: response.item.attributes,
+      references: response.item.references,
+    };
+  } catch (e) {
+    return {
+      status: 'error',
+      error: e.body || e.message,
+      id,
+    };
+  }
```

Comment:
Had to refactor this - for some reason, the old code wasn't catching errors being thrown in `contentManagement.client.get`. Now, no matter where the error is thrown, it should (hopefully) be caught 🫠